Frontend/Gmfctn/src/app/modules/+badges/badges.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { ProfileService } from 'src/app/core/services/profile-service/profile.service';
+import { Roles, User } from 'src/app/shared/models/user';
+import { defaultUser } from 'src/app/shared/models/dafault-user';
+
+@Component({
+  selector: 'app-badges',
+  templateUrl: './badges.component.html',
+  styleUrls: ['./badges.component.scss']
+})
+export class BadgesComponent implements OnInit, OnDestroy {
+  subscription$: Subscription = new Subscription();
+  height = 0;
+  margin = '';
+  user: User = defaultUser;
+  constructor(private profileService: ProfileService) {}
+
+  ngOnInit(): void {
+    this.calculateSize();
+    this.subscription$ = this.profileService.currentUser$.subscribe( user => this.user = user);
```

Comment:
Please, use this.subscription$.add(observable), as currently you're overriding the subscriptions instance with a subscription of the specific observable

Reply:
Fixed